lib/projectStructure.ts

```diff
@@ -1,36 +1,103 @@
-// @ts-nocheck
-const fs = require('fs');
-const path = require('path');
-const { walk } = require('@hubspot/local-dev-lib/fs');
-const { logger } = require('@hubspot/local-dev-lib/logger');
-const { logError } = require('./errorHandlers/index');
-
-const COMPONENT_TYPES = Object.freeze({
+import * as fs from 'fs';
+import * as path from 'path';
+import { walk } from '@hubspot/local-dev-lib/fs';
+import { logger } from '@hubspot/local-dev-lib/logger';
+import { logError } from './errorHandlers/index';
+
+type ComponentTypes = 'private-app' | 'public-app' | 'hubl-theme';
```

Comment:
Since you already declare the `COMPONENT_TYPES` object below, you could derive this type from that

`type ComponentTypes = ValueOf<typeof COMPONENT_TYPES>;`